arm-storage/2016-05-01/swagger/storage.json

```diff
@@ -0,0 +1,1301 @@
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets a list of key value pairs that describe the resource. These tags can be used for viewing and grouping this resource (across resource groups). A maximum of 15 tags can be provided for a resource. Each tag must have a key with a length no greater than 128 characters and a value with a length no greater than 256 characters."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/StorageAccountPropertiesCreateParameters"
+        }
+      },
+      "required": [
+        "sku",
+        "kind",
+        "location"
+      ],
+      "description": "The parameters used when creating a storage account."
+    },
+    "Endpoints": {
+      "properties": {
+        "blob": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Gets the blob endpoint."
+        },
+        "queue": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Gets the queue endpoint."
+        },
+        "table": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Gets the table endpoint."
+        },
+        "file": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Gets the file endpoint."
+        }
+      },
+      "description": "The URIs that are used to perform a retrieval of a public blob, queue, or table object."
+    },
+    "StorageAccountProperties": {
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Gets the status of the storage account at the time the operation was called.",
+          "enum": [
+            "Creating",
+            "ResolvingDNS",
+            "Succeeded"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": false
+          }
+        },
+        "primaryEndpoints": {
+          "$ref": "#/definitions/Endpoints",
+          "readOnly": true,
+          "description": "Gets the URLs that are used to perform a retrieval of a public blob, queue, or table object. Note that Standard_ZRS and Premium_LRS accounts only return the blob endpoint."
+        },
+        "primaryLocation": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Gets the location of the primary data center for the storage account."
+        },
+        "statusOfPrimary": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Gets the status indicating whether the primary location of the storage account is available or unavailable.",
+          "enum": [
+            "Available",
+            "Unavailable"
+          ],
+          "x-ms-enum": {
+            "name": "AccountStatus",
+            "modelAsString": false
+          }
```

Comment:
@amarzavery 
Seems unable to reply in your newly reply, so updated here, and replied inline.

1. This is a new api-version (2016-05-01) so how will you break customers?
[Jason] this is an incremental update of 2016-01-01. For the part we are discussing, they are shared with version 2016-01-01 for now. 

2. Fixing this in all the sdks is not the right solution.
[Jason]Yes. Fixing in swagger will reflect the changes to other sdks.

3. The storage spec, the way it is right now, is currently broken. Hardening the response from the server side is not a breaking change. In fact the response from the service will conform to the enum values specified in the swagger spec.

4. If you cannot fix the service then please fix the swagger spec. Please add all the variations in the enum that the server sends.
[Jason]Yes, fix should be in swagger spec. But not quite understand "the variations in the enum that the server sends"? Like following part?

        "statusOfPrimary": {
          "readOnly": true,
          "type": "string",
          "description": "Gets the status indicating whether the primary location of the storage account is available or unavailable.",
          "enum": [
            "available",
            "unavailable"
          ],
          "x-ms-enum": {
            "name": "AccountStatus",
            "modelAsString": false
          }
        },

Currently REST API docs on the doc website (docs.microsoft.com) stand incorrect because the swagger spec is incorrect. Please fix this ASAP.
[Jason] docs didn't reflect the enum part. And the description part is correct for "statusOfSecondary". For enum value in "StorageAccountKey", the value should be "Full" or "Read", will confirm that latter and I think it's better to fix in another PR.